### PR TITLE
feat(date-picker): content project the label, #8086

### LINKED ISSUE #8086: Make the date picker label consistent with the time picker's
## Is your feature request related to a problem? Please describe.
The date picker's label implementation should be made consistent with how the time picker's one is implemented.

## Describe the solution you'd like
It shouldn't be a property inside the component but a separate tag.

````html
<igx-date-picker>
    <label igxLabel>Custom label</label>
</igx-date-picker>
````

### PR DESCRIPTION
**Pull request by valadzhov**
Making the date picker's label consistent with that of the time picker, combo/select
Closes #8086 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ All notable changes for each version of this project will be documented in this
     - Allows for setting an input group `type` on a global level, so all input-group instances, including components using such an instance as a template will have their input group type set to the one specified by the token. It can be overridden on a component level by explicitly setting a `type`.
 - ` IgxExcelExporterService`
     - Added `worksheetName` property to the `IgxExcelExporterOptions`, that allows setting the name of the worksheet.
+- `IgxDatePicker`
+    - The `labelVisibility` has been removed and now a custom label is set by nesting a <label igxLabel></label> inside the <igx-date-picker><igx-date-picker> tags.
```
Comment (damyanpetev) [inline CHANGELOG.md:19]: Ummm has it been removed or deprecated?

File: projects/igniteui-angular/src/lib/date-picker/date-picker.component.html [hunk 2]
```diff
@@ -1,9 +1,15 @@
+<ng-template #labelTemplate>
+    <ng-content select="[igxLabel]"></ng-content>
+</ng-template>
+
 <ng-template #readOnlyDatePickerTemplate>
     <igx-input-group (click)="openDialog()">
+        <label igxLabel *ngIf="!labelDirective&&!labelVisibility">Date1</label>
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/date-picker/date-picker.component.html:7]: What is this label about? Where did Date1 come from?

File: projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts [hunk 3]
```diff
@@ -161,7 +163,15 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
      * ```html
      * <igx-date-picker [label]="Calendar"></igx-date-picker>
      * ```
+     * @deprecated By default the label is set to 'Date'. Use
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts:166]: By default the label is set to 'Date' is not necessary information for this property (the default value is visible). Put something like `Use igxLabel inside the date picker to change the label: <code>`

File: projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts [hunk 4]
```diff
@@ -161,7 +163,15 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
...
+     * ````html
+     * <igx-date-picker>
+     *      <label igxLabel>Custom label</label>
+     * </igx-date-picker>
+     * ````
+     * to set a custom label.
      */
+    @DeprecateProperty(`By default the label is set to 'Date'.`)
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts:174]: Same here - this is a console warning, really doesn't tell me much that the default value is Date :) Consider text similar to my above recommendation - "Use igxLabel.. "

File: projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts [hunk 5]
```diff
@@ -171,7 +181,18 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
...
      * <igx-date-picker [labelVisibility]="false"></igx-date-picker>
+     * @deprecated Use
+     * ````html
+     * <igx-date-picker>
+     *      <label igxLabel>Custom label</label>
+     * </igx-date-picker>
+     * ````
+     * to set a custom label.
      */
+    @DeprecateProperty(`Deprecated. Use
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts:192]: I really don't see how that replaces the visibility functionality. Perhaps this property shouldn't be deprecated yet?

File: projects/igniteui-angular/src/lib/grids/cell.component.html [hunk 6]
```diff
@@ -37,7 +37,7 @@
     </ng-container>
     <ng-container *ngIf="column.dataType === 'date'">
         <igx-date-picker [style.width.%]="100" [outlet]="grid.outlet" mode="dropdown"
-            [locale]="grid.locale" [(value)]="editValue" [igxFocus]="true" [labelVisibility]="false">
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/grids/cell.component.html:40]: I assume those are hidden for a reason... and can't just be removed like that

**Review by damyanpetev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (valadzhov) [reply CHANGELOG.md:19]: It's been deprecated.

**Review by Lipata — APPROVED** [review]

**Pull request merged**